stl/inc/numeric

```diff
@@ -192,32 +202,42 @@ inline constexpr bool _Default_ops_transform_reduce_v = false;
 #endif // _STD_VECTORIZE_WITH_FLOAT_CONTROL
 
 template <class _InIt1, class _InIt2, class _Ty, class _BinOp1, class _BinOp2>
-_NODISCARD _Ty transform_reduce(
+_NODISCARD _CONSTEXPR20_ICE _Ty transform_reduce(
     _InIt1 _First1, _InIt1 _Last1, _InIt2 _First2, _Ty _Val, _BinOp1 _Reduce_op, _BinOp2 _Transform_op) {
     // return commutative and associative transform-reduction of sequences, using
     // _Reduce_op and _Transform_op
     _Adl_verify_range(_First1, _Last1);
     auto _UFirst1      = _Get_unwrapped(_First1);
     const auto _ULast1 = _Get_unwrapped(_Last1);
     auto _UFirst2      = _Get_unwrapped_n(_First2, _Idl_distance<_InIt1>(_UFirst1, _ULast1));
-    if constexpr (_Default_ops_transform_reduce_v<_Unwrapped_t<const _InIt1&>, _Unwrapped_t<const _InIt2&>, _Ty,
-                      _BinOp1, _BinOp2>) {
-        (void) _Reduce_op; // TRANSITION, VSO-486357
-        (void) _Transform_op; // TRANSITION, VSO-486357
-        return _Transform_reduce_arithmetic_defaults(_UFirst1, _ULast1, _UFirst2, _STD move(_Val));
-    } else {
+#ifdef __cpp_lib_is_constant_evaluated
+    if (_STD is_constant_evaluated()) {

```

Comment:
Ditto "Why use `is_constant_evaluated` here instead of `_NODISCARD constexpr _Ty _Transform_reduce_arithmetic_defaults`?

Reply:
DevCom-878972